packages/grafana-ui/src/components/uPlot/config/UPlotSeriesBuilder.ts

```diff
@@ -146,16 +148,15 @@ let builders: PathBuilders | undefined = undefined;
 function mapDrawStyleToPathBuilder(
   style: DrawStyle,
   lineInterpolation?: LineInterpolation,
-  opts?: any
+  barAlignment?: BarAlignment
 ): Series.PathBuilder {
-  // This should be global static, but Jest initalization was failing so we lazy load to avoid the issue
+  const barWidthFactor = 0.6;
+  const barMaxWidth = Infinity;
   if (!builders) {
+    // This should be global static, but Jest initalization was failing so we lazy load to avoid the issue
     const pathBuilders = uPlot.paths;
-    const barWidthFactor = 0.6;
-    const barMaxWidth = Infinity;
 
     builders = {
-      bars: pathBuilders.bars!({ size: [barWidthFactor, barMaxWidth] }),
```

Comment:
@dprokop the point of having them cached here is so they don't need to be re-initialized on ever redraw() call (like during resize). please do the same thing here as the stepBefore & stepAfter. make something like `barCenter`, `barBefore`, `barAfter`

Reply:
This one cannot be cached, at least not now, as it depends on the position config which is dynamic as compared to width factor and max width.

Reply:
i'm confused. how is this any different than what we do with stepBefore/stepAfter and lineInterpolation?

Reply:
![image](https://user-images.githubusercontent.com/43234/105393712-47300d00-5be2-11eb-98a6-644d0bb213a3.png)

Reply:
i also tried pulling that whole block up and out in https://github.com/grafana/grafana/pull/30518, but that's still failing Jest/CI for mysterious reasons :(

Reply:
@leeoniya oh shit, u totally right with the separate builder for bar types, rather than single one that I have proposed. It makes sense- thans!